rinja_derive/src/generator.rs

```diff
@@ -558,85 +568,92 @@ impl<'a> Generator<'a> {
             self.handle_ws(ws_before);
         }
 
-        for (pos, cond_info) in conds.conds.iter().enumerate() {
+        let mut iter = conds.conds.iter().enumerate().peekable();
+        while let Some((pos, cond_info)) = iter.next() {
             let cond = cond_info.cond;
 
-            self.handle_ws(cond.ws);
-            flushed += self.write_buf_writable(ctx, buf)?;
-            if pos > 0 {
-                self.locals.pop();
```

Comment:
I like that this line is gone. Looks extremely error-prone.

Reply:
Yes, I don't like states that you need to not forget about changing yourself. I'm very fan of callback approach for this reason (even in other language like C and C++).